velox/common/fuzzer/ConstrainedGenerators.h

```diff
@@ -375,4 +388,52 @@ class JsonInputGenerator : public AbstractInputGenerator {
 
   folly::json::serialization_opts opts_;
 };
+
+/// Generates a JSON path for JSON of a given type.
+/// @param jsonType The type of data represented by the JSON.
+/// @param mapKeys Candidate key values of maps in the JSON. All maps in the
+/// JSON are assumed to share the same key type and candidate key values.
+/// @param maxContainerLength The maximum length of a container (array or map)
+/// in the JSON.
+/// @param makeRandomVariation If true, the generator will generate JSON path
+/// not supported by JsonExtractor.
+/// This generator generates the following JSON paths:
+///  - On root: $
+///  - On strings: .length()
+///  - On arrays: .length(), [index], [], [begin:end:step]
+///  - On objects: .length(), [key], ['key'], [], [*], .key
```

Comment:
how are keys generated?
There are some symbols like `:` and `,` for this path type `[key]` which should not be valid and other symbols for `.key` which are not valid. As a follow up we can choose to include those to cover unsupported cases.

Reply:
A list of candidate keys are passed to `JsonPathGenerator` and `JsonPathGenerator` just randomly select one key from the list. `JsonPathGenerator` can assume that the JSON-string used together with the generated JSON path also only use these candidate keys in objects.

I don't fully understand your comment about symbols like `:` and `,` in `[key]`. Do you mean `key` can contain `:` or `,` and that should be invalid?

Reply:
> I don't fully understand your comment about symbols like : and , in [key]. Do you mean key can contain : or , and that should be invalid?

For eg.
`$.k:y` is valid and is equivalent to `$.["k:y"]`
however,
`$.[k:y]` is not valid